stdlib/LinearAlgebra/src/tridiag.jl

```diff
@@ -723,6 +723,62 @@ end
 Base._sum(A::Tridiagonal, ::Colon) = sum(A.d) + sum(A.dl) + sum(A.du)
 Base._sum(A::SymTridiagonal, ::Colon) = sum(A.dv) + 2sum(A.ev)
 
+function Base._sum(A::Tridiagonal, dims::Integer)
+    res = Base.reducedim_initarray(A, dims, zero(eltype(A)))
+    n = length(A.d)
+    @inbounds begin
+        if dims == 1
+            res[1] = A.dl[1] + A.d[1]
+            for i = 2:n-1
+                res[i] = A.dl[i] + A.d[i] + A.du[i-1]
+            end
+            res[n] = A.d[n] + A.du[n-1]
+        elseif dims == 2
+            res[1] = A.d[1] + A.du[1]
+            for i = 2:n-1
+                res[i] = A.dl[i-1] + A.d[i] + A.du[i]
+            end
+            res[n] = A.dl[n-1] + A.d[n]
+        elseif dims >= 3
+            for i = 1:n-1
+                res[i,i+1] = A.du[i]
+                res[i,i]   = A.d[i]
+                res[i+1,i] = A.dl[i]
+            end
+            res[n,n] = A.d[n]
+        end
+    end
+    res
+end
+
+function Base._sum(A::SymTridiagonal, dims::Integer)
+    res = Base.reducedim_initarray(A, dims, zero(eltype(A)))
+    n = length(A.dv)
+    @inbounds begin
+        if dims == 1
+            res[1] = A.ev[1] + A.dv[1]
```

Comment:
I guess this can cause segfault for cases like `SymTridiagonal(Int[], Int[])` and `SymTridiagonal([1], Int[])`? Apparently they are constructable...  I guess you need special handling for those cases?

Similar caution may be required for other matrix types.

Reply:
You're right, thanks for the hint. I will add additional checks in the new methods of `_sum`.

Reply:
I also had to add new checks in some conversion methods to `Matrix`. Interestingly, I didn't get errors while running the tests locally in the REPL. Is there a way to disable `@inbounds` (i.e. enforce bounds checking everywhere) in the REPL?

Reply:
AFAIK you need to restart `julia` with `--check-bounds=yes`.

Reply:
Okay, thanks.